cubical/solver.py

```diff
@@ -710,6 +727,8 @@ def run(self):
                      self.vdm.weighted_obser, self.vdm.weighted_model, self.vdm.flags_arr,
                      self.sol_opts, label=self.label)
 
+        self.output_weights = self.gm.output_weights
```

Comment:
Is this safe? I have no intuition for this.

Reply:
It is safe, I added this because the return statement in _solver_gains was change. So this is the only way return and save the weights from the robust solver in the MS